spring-data-cassandra/src/main/java/org/springframework/data/cassandra/repository/query/AbstractCassandraQuery.java

```diff
@@ -194,18 +178,21 @@ protected void warnIfMoreResults(Iterator<Row> iterator) {
 	}
 
 	public ConversionService getConversionService() {
-		return conversionService;
+		return template.getConverter().getConversionService();
 	}
 
+	/**
+	 * @param conversionService
+	 * @deprecated {@link org.springframework.data.cassandra.mapping.CassandraMappingContext} handles type conversion.
+	 */
+	@Deprecated
 	public void setConversionService(ConversionService conversionService) {
-
-		Assert.notNull(conversionService);
-		this.conversionService = conversionService;
+		// not supported anymore
```

Comment:
Maybe throw an `UnsupportedOperationException`.  I think maybe it is better to fail fast here then to leave the user wondering why their (custom) `ConversionService` (of old) was not used.